isis/src/galileo/apps/gllssical/main.cpp

```diff
@@ -437,30 +439,52 @@ void calculateScaleFactor0(Cube *icube, Cube *gaincube) {
 
   if (iof) {
     Pvl *label = icube->label();
-    Spice spicegll(*icube);
-    spicegll.instrumentPosition()->SetAberrationCorrection("LT+S");
     QString startTime = label->findGroup("Instrument",Pvl::Traverse)["SpacecraftClockStartCount"][0];
-    Isis::FileName sclk(label->findGroup("Kernels",Pvl::Traverse)["SpacecraftClock"][0]);
-    QString sclkName(sclk.expanded());
-    furnsh_c(sclkName.toLatin1().data());
     double obsStartTime;
-    scs2e_c(-77, startTime.toLatin1().data(), &obsStartTime);
-    spicegll.setTime(obsStartTime);
-    double sunv[3];
-    spicegll.sunPosition(sunv);
 
-    double sunkm = vnorm_c(sunv);
-
-    //  Convert to AU units
-    rsun = sunkm / 1.49597870691E8 / 5.2;
-
-    /*
-     * We are calculating I/F, so scaleFactor0 is:
-     *
-     *         S1       K
-     *      -------- * --- (D/5.2)**2
-     *         A1       Ko
-     */
+    try {
+      Camera *cam;
+      cam = icube->camera();
+      cam->instrumentPosition()->SetAberrationCorrection("LT+S");
+      obsStartTime = cam->getClockTime(startTime.toLatin1().data(), -77).Et();
+      cam->setTime(obsStartTime);
+
+      // rsun converted to AU
+      rsun = cam->sunToBodyDist() / 1.49597870691E8 / 5.2;
+    } 
+    catch (IException &e) {
+      // try original fallback for previously spiceinited data 
```

Comment:
Previously/old/prior to this update spiceinited data doesn't have all the information it needs to go through the camera.